src/borg/archive.py

```diff
@@ -1065,9 +1065,25 @@ def verify_data(self):
                 logger.warning('Found defect chunks. They will be deleted now, so affected files can '
                                'get repaired now and maybe healed later.')
                 for defect_chunk in defect_chunks:
-                    del self.chunks[defect_chunk]
-                    self.repository.delete(defect_chunk)
-                    logger.warning('chunk %s deleted.', bin_to_hex(defect_chunk))
+                    DEFECT_CHUNK_RETRIES = 3
+                    retries_failed = 0
+                    for retry in range(DEFECT_CHUNK_RETRIES):
+                        if retry == DEFECT_CHUNK_RETRIES - 1:
+                            # last retry, wait a little, maybe it helps
+                            time.sleep(1)
+                        encrypted_data = self.repository.get(defect_chunk)
+                        try:
+                            _chunk_id = None if defect_chunk == Manifest.MANIFEST_ID else defect_chunk
+                            self.key.decrypt(_chunk_id, encrypted_data)
+                        except IntegrityError:
+                            retries_failed += 1
+                    if retries_failed == DEFECT_CHUNK_RETRIES:
+                        # all retries failed, get rid of this chunk
+                        del self.chunks[defect_chunk]
+                        self.repository.delete(defect_chunk)
+                        logger.warning('chunk %s deleted.', bin_to_hex(defect_chunk))
```

Comment:
IMHO the usual case is the "really broken chunk", not chunks corrupted by a bad cable or so. Therefore I think it'd be best to log only the exceptional case (chunk recovered, won't be deleted), not the announced one (chunk deleted). Logging the chunk ID isn't necessary, since they were all logged as errors already.